ibis/omniscidb/operations.py

```diff
@@ -313,6 +313,74 @@ def extract_field_formatter(translator, expr, sql_attr=sql_attr):
     return extract_field_formatter
 
 
+def _n_weeks_year(year: ir.NumericValue):
+    """
+    Return the number of the total of weeks for the given year.
+
+    Parameters
+    ----------
+    year : ibis.expr.types.NumericValue
+
+    Returns
+    -------
+    Ibis.expr.types.Expr
+
+    See Also
+    --------
+    https://en.wikipedia.org/wiki/ISO_week_date
+
+    """
+    # adjustment factor for week year extraction
+    def adjustment_factor(year):
+        return (year + (year // 4) - (year // 100) + (year // 400)) % 7
+
+    need_adjustment = (adjustment_factor(year) == 4) | (
+        adjustment_factor(year - 1) == 3
+    )
+
+    return 52 + (ibis.case().when(need_adjustment, 1).else_(0).end())
+
+
+def _woy_preliminary(d: Union[ir.DateValue, ir.TimestampValue]) -> ir.Expr:
```

Comment:
Is a function really needed for this? This is being used just once, feels like moving this two lines of code inside `_extract_woy` would keep things much simpler.